### PR TITLE
Add page number loging to Ariba script

### PR DESCRIPTION
**Pull request by milospb**

### CODE REVIEW

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js [hunk 1]
```diff
@@ -1,399 +1,410 @@
...
-			}
-		}
-	}
-	sync.dataStore.save('pendingApprovablesRequisitions', rqUserList) //saving approvers
-	return groups
+    //this function is resolving approvers for requisitions and copy necessary informations to create notifications
+    let rqUserList = []
+    for (let appReqsr of json) {
+        for (let approvers of appReqsr.approvers) {
+            if (approvers.type !== undefined) {
```
Comment (vojtacitrite) [inline bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:58]: `null` is allowed? what about other types? what about something like this: `if (['user', 'group'].includees(approvers.type)) {`

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js [hunk 2]
```diff
@@ -1,399 +1,410 @@
...
+                        approvers = flatJsonObject(approvers, 'user', '')
+                        rqUserList.push(approvers)
+                    }
+                    if (approvers.type === 'group') { //if approver is a group
+                        let groupMembers = await resolveGroupMembers(sync, approvers.id, groups);
+                        rqUserList = rqUserList.concat(resolveGroupApproval(approvers, groupMembers, approvers.id))
+                    }
+                }
+                catch (e) {
+                    console.log('Warning: unable to get all requisition data for user ' + e.message)
```
Comment (vojtacitrite) [inline bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:76]: ```console.log(`Warning: unable to get all requisition data for ${approvers.type}: ${e.message}`)```

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js [hunk 3]
```diff
@@ -1,399 +1,410 @@
...
+        return groups.get(groupId);
+    }
+    let responseData = await sync.client.fetch('groups/' + groupId + '/members?realm=' + sync.integrationParameters.realm + '&limit=1000', {
+        headers: {
+            'apikey': sync.integrationParameters.apiKey
+        }
+    });
+
+    if (responseData.ok) {
+        let memberList = (await responseData.json()).map(member => {
```
Comment (vojtacitrite) [inline bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:97]: just tip: could be solved by Lodash ```.map(member => _.pick(member, ['emailAdress', 'phone', etc ]))```
or by expression ```.map(({emailAddress, phone, etc}) => ({emailAddress, phone, etc}))```

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js [hunk 4]
```diff
@@ -1,399 +1,410 @@
...
-			user.totalCostCurrency = json.totalCostCurrency
-		}
-		catch (e) {
-			console.log('Warning: unable to get all requisition data for user:' + user)
-		}
-	}
-	return userList
+function resolveGroupApproval(json, groupMembers, groupId) {
+    //currently we are able to store only flat json object, so all nested object must be flattened
+    return groupMembers.map(member => {
```
Comment (vojtacitrite) [inline bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:120]: the same tip as for `resolveGroupMembers`

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js [hunk 5]
```diff
@@ -1,399 +1,410 @@
...
+                }
+                sync.dataStore.save('rq_lineitems', flatLineItems(json.lineItems)) //saving line items
+                delete json.lineItems
+            }
+            sync.dataStore.save('requisition_details', flatRequisition(json))
+        }
+    }
 }
 
 function parseApprovablesGetDocIds(json, documentsLists) {
```
Comment (vojtacitrite) [inline bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:173]: what is actually `json` ? it's some array of some items but definitely not an object nor a JSON

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js [hunk 6]
```diff
@@ -1,399 +1,410 @@
...
-	for (let i = 0; i < json.length; i++) {
-		if (json[i].documentType == 'ariba.purchasing.core.Requisition') {
-			documentsLists.requisitions.push(json[i].uniqueName)
-		}
-		if (json[i].documentType == 'ariba.invoicing.core.Invoice') {
-			documentsLists.invoices.push(json[i].uniqueName)
-		}
-	}
-	return documentsLists
+    for (let i = 0; i < json.length; i++) {
```
Comment (vojtacitrite) [inline bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:174]: `for` was already used before this PR, so just a tip: could be replaced by `map` or `foreEeach`

**Review by vojtacitrite — CHANGES_REQUESTED** [review]

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js (hunk 4 above)
Comment (milospb) [reply bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:120]: Will not update

**Review by vojtacitrite — APPROVED** [review]

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js (hunk 6 above)
Comment (milospb) [reply bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:174]: will not updated

File: bundles/http/Citrix/SAP-Ariba/Ariba_v2.js (hunk 1 above)
Comment (milospb) [reply bundles/http/Citrix/SAP-Ariba/Ariba_v2.js:58]: fixed

**Pull request merged**